src/vis/bar/utils.ts

```diff
@@ -0,0 +1,240 @@
+            formList: ['groupBy', 'barMultiplesBy', 'barDirection', 'barGroupType', 'barNormalized']
+        };
+    }
+
+    const catCols: CategoricalColumn[] = columns.filter((c) => config.catColumnsSelected.filter((d) => c.info.id === d.id).length > 0 && c.type === EColumnTypes.CATEGORICAL) as CategoricalColumn[];
+    const vertFlag = config.direction === EBarDirection.VERTICAL;
+    const normalizedFlag = config.display === EBarDisplayType.NORMALIZED;
+    const plots: PlotlyData[] = [];
+
+    if(catCols.length > 0) {
+        const catCurr = catCols[0];
+        if(config.group && config.multiples) {
+
+            const currGroupColumn = getCol(columns, config.group) as CategoricalColumn;
+            const currMultiplesColumn = getCol(columns, config.multiples) as CategoricalColumn;
+
+
+            const uniqueGroupVals = [...new Set(currGroupColumn.vals.map((v) => v.val))];
+            const uniqueMultiplesVals = [...new Set(currMultiplesColumn.vals.map((v) => v.val))];
+
+            const uniqueColVals = [...new Set(catCurr.vals.map((v) => v.val))];
+
+            uniqueMultiplesVals.forEach((uniqueMultiples) => {
+                uniqueGroupVals.forEach((uniqueGroup) => {
+
+                    const groupedLength = uniqueColVals.map((v) => {
+                        const allObjs = catCurr.vals.filter((c) => c.val === v).map((c) => c.id);
+                        const allGroupObjs = currGroupColumn.vals.filter((c) => c.val === uniqueGroup).map((c) => c.id);
+                        const allMultiplesObjs = currMultiplesColumn.vals.filter((c) => c.val === uniqueMultiples).map((c) => c.id);
+
+                        const joinedObjs = allObjs.filter((c) => allGroupObjs.includes(c) && allMultiplesObjs.includes(c));
+
+                        return normalizedFlag ? joinedObjs.length / allObjs.length * 100 : joinedObjs.length;
+                    });
+
+                    plots.push({
+                        data: {
+                            x: vertFlag ? [...new Set(catCurr.vals.map((v) => v.val))] : groupedLength,
+                            y: !vertFlag ? [...new Set(catCurr.vals.map((v) => v.val))] : groupedLength,
+                            orientation: vertFlag ? 'v' : 'h',
+                            xaxis: counter === 1 ? 'x' : 'x' + counter,
+                            yaxis: counter === 1 ? 'y' : 'y' + counter,
+                            showlegend: counter === 1 ? true : false,
+                            type: 'bar',
+                            name: uniqueGroup,
+                            marker: {
+                                color: scales.color(uniqueGroup),
+                            }
+                        },
+                        xLabel: vertFlag ? catCurr.info.name : normalizedFlag ? 'Percent of Total' : 'Count',
+                        yLabel: vertFlag ? normalizedFlag ? 'Percent of Total' : 'Count' : catCurr.info.name
```

Comment:
Here are also some language strings

Reply:
Since I imagine the way we do this will change in the near future, leaving any implementation of i18n to future PRs.